server/models/Collective.js

```diff
@@ -1270,11 +1270,29 @@ export default function(Sequelize, DataTypes) {
     .then(result => Promise.resolve(-parseInt(result.toJSON().total, 10)));
   };
 
-  // Get the total amount spent last month (burn rate)
-  Collective.prototype.getTotalAmountSpentLastMonth = function() {
-    const lastMonth = new Date;
-    lastMonth.setMonth(lastMonth.getMonth() - 1);
-    return this.getTotalAmountSpent(lastMonth);
+  // Get the average monthly spending based on last 90 days
+  Collective.prototype.getMonthlySpending = function() {
```

Comment:
Can this reuse the raw query so we don't have repeated raw sql around the codebase? 

```es6
import { getCollectivesOrderedByMonthlySpending } from '../lib/queries';

// in Collective.prototype.getMonthlySpending
const { collectives } = getCollectivesOrderedByMonthlySpending({ where: { id: this.id }});
return collectives[0].monthlySpending;
```

Reply:
Mmmm... I see your point. But the problem is that one query is supposed to return a list of collectives whereas the other is supposed to return the actual average monthly spending for a given collective. I could consolidate both queries and make a condition that if `limit===1` we return directly the number? And if limit > 1 we return the total numbers of results and the current page of results? Wdyt?

Reply:
Why make a condition in the raw query if you can pass limit 1 and where the id = the collective id? You're using the raw query inside the `getMonthlySpending` method of the Collective model, so the raw query doesn't need to be concerned with how the returned result is used.

Reply:
Because it seems like a waste to make two queries to get number of results, then the result for that second case.

Reply:
The queries are performed parallel, so it feels like a premature optimization to avoid a single `count` query in favor of changing the return value of a reusable query.

Reply:
Except there is no loaders for computing that `monthlySpending` yet so the query is run for every result of a list. So when when we fetch 10 collectives, we are doing 20 queries instead of 10 to get that value for each.

(that said, I still think that you have a valid point. I'll adapt the query accordingly. One quick small optimization is to not perform the count query if there is a condition on `Collective.id`)

Thanks for the constructive exchange here. It feels like we are improving the code! :-)